src/layout/NavigationLink.js

```diff
@@ -1,23 +1,35 @@
-// file: shared/layout/NavigationLink.js
+// file: src/layout/NavigationLink.js
 
 import React, { PropTypes } from 'react';
 import classNames from 'classnames';
-import { Link } from 'react-router';
 
 export class NavigationLink extends React.Component {
   static propTypes = {
+    id: PropTypes.string,
```

Comment:
I'm not really sure about why we need to specify `id` in propTypes. 

Let's skip `id` and `href`, and declare only `className` and `tag`. Then you just pass `href` or `to` depending on whether you use `'a'` or `NavigationLink` as tag. Moreover, `...otherProps` will handle the job properly.